backend/data/users.js

```diff
@@ -301,6 +301,27 @@ async function addGrade(studentId, courseId, assignmentId, grade) {
   return { gradeAdded: true };
 }
 
+/**
+ * Fetch grade of a student by assignmentId
+ * @param {string} assignmentId 
+ * @param {string} studentId 
+ * @returns object
+ */
+async function fetchGrade(assignmentId, studentId) {
+  const errorMSg = 'No assignments found for the given assignment id';
+  const user = await getUser(studentId);
+  if (user && !user.classes) throw new Error(errorMSg);
```

Comment:
Might be good to add a check for if(!!user) with a throw, just to cover all bases

Reply:
Thanks, @karl-olsen I have added the changes in the latest PR